src/Components/Web.JS/src/Rendering/StreamingRendering.ts

```diff
@@ -36,7 +36,14 @@ class BlazorStreamingUpdate extends HTMLElement {
       if (node instanceof HTMLTemplateElement) {
         const componentId = node.getAttribute('blazor-component-id');
         if (componentId) {
-          insertStreamingContentIntoDocument(componentId, node.content);
+          // For enhanced nav page loads, we automatically cancel the response stream if another enhanced nav supersedes it. But there's
+          // no way to cancel the original page load. So, to avoid continuing to process <blazor-ssr> blocks from the original page load
+          // if an enhanced nav supersedes it, we must explicitly check whether this content is from the original page load, and if so,
+          // ignore it if any enhanced nav has started yet. Fixes https://github.com/dotnet/aspnetcore/issues/50733
+          const isFromEnhancedNav = node.getAttribute('enhanced-nav') === 'true';
+          if (isFromEnhancedNav || hasNeverStartedAnyEnhancedPageLoad()) {
+            insertStreamingContentIntoDocument(componentId, node.content);
+          }
```

Comment:
Can it happen that we cancel an enhanced nav mid-way (while some content has already been inserted) and we start processing a new enhanced nav response? Would we be in a situation where we've updated half the DOM and we start updating the rest with a newer version? Or will the new version replace the updated bits so far + the non-updated bits.

Reply:
> Or will the new version replace the updated bits so far + the non-updated bits.

Yes, that. Each enhanced nav response is treated as a desired end state, and we diff as needed to reach that state, no matter what state we were in before.

Because of the single-threadedness of JS, and because we cancel any preceding enhanced nav response stream before we issue a new request, only the latest enhanced nav response is processed and it full replaces whatever was there before. If the previous request was streaming SSR and hadn't finished sending streaming updates, we don't care - we just cancel the previous request and hence won't see any more of its streaming updates.

Reply:
@SteveSandersonMS could that behavior cause weird outcomes/bugs? I imagine that worst case scenario it ends up with the whole document being replaced, I am just trying to think of situations where it could be problematic break things.

Reply:
Not that I know of. This is the way it's designed and expected to work. The guiding principle of enhanced nav is that it should produce the same outcome as a full-page navigation, with the exception of specific things we explicitly preserve (interactive components and `data-preserve` elements). On those grounds it's correct to disregard orphaned streaming updates from prior navigations.